store/src/state.rs

```diff
@@ -342,27 +342,35 @@ impl State {
         account_ids: &[AccountId],
         note_tag_prefixes: &[u32],
         nullifier_prefixes: &[u32],
-    ) -> Result<(StateSyncUpdate, MmrDelta, MerklePath), anyhow::Error> {
+    ) -> Result<(StateSyncUpdate, MmrDelta), anyhow::Error> {
         let inner = self.inner.read().await;
 
         let state_sync = self
             .db
             .get_state_sync(block_num, account_ids, note_tag_prefixes, nullifier_prefixes)
             .await?;
 
-        let (delta, path) = {
-            let delta = inner
-                .chain_mmr
-                .get_delta(block_num as usize, state_sync.block_header.block_num as usize)?;
-
-            let proof = inner
-                .chain_mmr
-                .open(block_num as usize, state_sync.block_header.block_num as usize)?;
-
-            (delta, proof.merkle_path)
+        let delta = if block_num == state_sync.block_header.block_num {
+            // The client is in sync with the chain tip.
+            MmrDelta {
+                forest: block_num as usize,
+                data: vec![],
+            }
+        } else {
+            // Important notes about the boundary conditions:
+            //
+            // - The Mmr forest is 1-indexed whereas the block number is 0-indexed. The Mmr root
+            // contained in the block header always lag behind by one block, this is because the Mmr
+            // leaves are hashes of block headers, and we can't have self-referential hashes. These two
+            // points cancel out and don't require adjusting.
+            // - Mmr::get_delta is inclusive, whereas the sync_state request block_num is defined to be
+            // exclusive, so the from_forest has to be adjusted with a +1
+            let from_forest = (block_num + 1) as usize;
+            let to_forest = state_sync.block_header.block_num as usize;
+            inner.chain_mmr.get_delta(from_forest, to_forest)?
         };
```

Comment:
Should we also add a check here handling potential malicious input? Specifically, what will happen here if `block_num` is much bigger than `state_sync.block_header.block_num`? We should probably catch this and return an error.

Reply:
Or I guess this will happen naturally inside `ChainMmr::get_delta()`?

Reply:
> Or I guess this will happen naturally inside ChainMmr::get_delta()?

Yep